### DISCUSSION TITLE
frame_rate and hours in SMPTE offset message

### DISCUSSION THREAD

**Discussion by bixb922** [mido/mido#338] [General]
Hello,
I believe this is rather low priority. In meta.py it says:

    def decode(self, message, data):
        message.frame_rate = _smpte_framerate_decode[(data[0] >> 6)]
        message.hours = (data[0] & 0x3f)

From midi.org I understand the midi standard as follows for the frame rate/hours byte of the SMPTE message:
(https://www.midi.org/forum/45-smpte-offset-meta-event-smf, https://en.wikipedia.org/wiki/MIDI_timecode) 

0 yy zzzzz (8 bits)

0 is a reserved field (No other value is allowed)
yy is the time code type (FPS)
zzzzz is the hour count (0-23)

So the code in meta.py should probably be:

    def decode(self, message, data):
        message.frame_rate = _smpte_framerate_decode[(data[0] >> 5)]
        message.hours = (data[0] & 0x1f)

Best regards,
Hermann

**Comment by rdoursenaud** [comment]
As someone who has some experience digging in the arcanes of MIDI Time code implementation, I can attest that your assessment is correct!
Care to do a pull request?
If not, I volunteer to cook one.
